### PR TITLE
Expose payment amount limits

### PR DESCRIPTION
**Pull request by gcomte**
Expose limits that exist for receiving payments:

- Max amount to receive per payment during beta phase
- Max amount to receive without being charged a setup fee
- Mandatory min amount to receive (to cover setup fees)

### CODE REVIEW

File: src/lipalightninglib.udl [hunk 1]
```diff
@@ -227,6 +227,22 @@ dictionary LspFee {
...
+dictionary ChannelRelatedLimit {
+    AmountLimitType limit_type;
+    u64 amount_sat;
+};
+
+enum AmountLimitType {
+    "MaxFreeReceive", // Soft limit: The maximum sat amount a user can receive without being charged a setup fee
+    "MinReceive", // Hard limit: The minimum sat amount a user must receive with the next payment
+        // If this limit is provided, that means a setup fee will be charged for the incoming payment
+};
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:244]: Would it make sense to have an enum for the channel-based limit?

For example:
```suggestion
dictionary PaymentAmountLimits {
    u64 max_receive_sat; // Hard limit: The maximum sat amount a user is allowed to receive per payment
    ChannelRelatedLimit channel_related_limit;
};

[Enum]
interface ChannelRelatedLimit {
    NoLimit(); // inbound capacity >= max_receive_sat
    MaxFreeReceive(u64 limit);
    MinReceive(u64 limit);
};
```

File: eel/tests/receiving_payments_test.rs [hunk 2]
```diff
@@ -97,6 +111,30 @@ mod receiving_payments_test {
...
 
+            let channels_info = node.get_node_info().channels_info;
+            assert_moderate_inbound_capacity(&node, channels_info.inbound_capacity_msat);
+            assert_eq!(channels_info.num_usable_channels, 2);
+            assert_eq!(
+                channels_info.total_channel_capacities_msat - capacity_first_channel,
+                600_000_000
+            ); // new channel has a 0.6M sats capacity
+            assert_eq!(channels_info.local_balance_msat, 498_001_000); // Paid 2k for LSP, but had already 1 sat
+            info!("Restarting node..."); // to test that the graph is persisted and retrieved correctly
```
Comment (danielgranhao) [inline eel/tests/receiving_payments_test.rs:122]:
```suggestion
            info!("Restarting node...");
```

File: src/lipalightninglib.udl (hunk 1 above)
Comment (gcomte) [reply src/lipalightninglib.udl:244]: Yes, that's probably the more idiomatic way

Comment (danielgranhao) [reply src/lipalightninglib.udl:244]: Maybe `ChannelRelatedLimit` could even be `LiquidityLimit`, but I'm fine with the current name if you prefer it :)

Comment (gcomte) [reply src/lipalightninglib.udl]: Yes I like `LiquidityLimit` better. Couldn't find a good term myself. Thanks!

Check out 49546d2b53d16fde7b260b9861a7c1d480f4d915

File: eel/src/payment.rs [hunk 3]
```diff
@@ -80,3 +82,39 @@ impl Payment {
...
 }
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct PaymentAmountLimits {
+    pub max_receive_sat: u64,
+    pub liquidity_limit: LiquidityLimit,
+}
+
+impl PaymentAmountLimits {
+    pub fn fetch(inbound_capacity: u64, lsp_min_fee: u64) -> Self {
```
[the remark this answers is not in the document]
Comment (gcomte) [reply eel/src/payment.rs:93]: How about `obtain` or `retrieve`?

I agree that `fetching` sounds like it involves some significant roundtrip, but on the other hand `building` sounds like some kind of allocation happening that's way bigger than what's happening here.

File: eel/src/payment.rs [hunk 4]
```diff
@@ -80,3 +82,39 @@ impl Payment {
...
+        let min_receive_amount = lsp_min_fee * 2;
```
[the remark this answers is not in the document]
Comment (gcomte) [reply eel/src/payment.rs]: be2b716ae3e56cc02c1c4639b8a6b772c41f7b1a

File: eel/src/payment.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielgranhao) [reply eel/src/payment.rs:94]: Hmm, once again, those terms seem to indicate that the info isn't local. What about just a simple constructor `new()`?

File: eel/src/payment.rs [hunk 5]
```diff
@@ -80,3 +83,39 @@ impl Payment {
         false
     }
 }
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct PaymentAmountLimits {
+    pub max_receive_sat: u64,
+    pub liquidity_limit: LiquidityLimit,
```
[the remark this answers is not in the document]
Comment (gcomte) [reply eel/src/payment.rs]: Does not work

**Review by danielgranhao — APPROVED** [review]

File: eel/src/payment.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (gcomte) [reply eel/src/payment.rs]: I ended up changing it to `calculate()`.

**Pull request merged**